src/java/org/apache/cassandra/auth/AuthenticatedUser.java

```diff
@@ -74,6 +105,24 @@ public RoleResource getPrimaryRole()
         return role;
     }
 
+    /**
+     * The mode of authentication used to authenticate this user, this will usually be
+     * some value of {@link AuthenticationMode#getDisplayName()} unless
+     * an implementor provides their own custom authentication scheme.
+     */
+    public String getAuthenticationMode()
+    {
+        return authenticationMode;
```

Comment:
@tolbertam I think that `private final String authenticationMode;` can be of type `AuthenticationMode` and the only place where it needs to be String (here) you would call `displayName()` on that. I would just use AuthenticationMode everywhere I could and converted that to String when necessary.

Reply:
this is a non-issue with the PR I provided in the comment above.

Reply:
> because 3rd party implementation with their own mode would not be in AuthenticationMode enum so they still need to find a way how to populate this with something custom.

Exactly, I think the solution you came up works really well for this.  We get a type with some well-defined values, and third-party implementors can add their own if they'd like.